lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js

```diff
@@ -15,67 +13,42 @@ class FeatureCard extends React.Component {
   constructor(props) {
     super(props);
 
-    // These variables shouldn't force a state change & subsequent render
-    this.requests = [];
-    this.campaigns = [];
-    this.campaignsToExclude = setting('dsUser.activity.signups', []);
-    this.campaignsApiPage = 1;
-    this.apiClient = new ApiClient('v1');
-
     this.state = {
       actionDisabled: false,
       campaignIndex: -1,
       isSignedUp: false,
       isLoading: false,
     };
 
+    this.campaigns = [];
+    this.controller = new GalleryController();
```

Comment:
It would also help limit what the FeatureCard has to do... all it really cares about is receiving some info about what campaign to display and allowing a passed in signup action (along with a viewmore action when it's a single one) instead of dealing with gathering all the campaigns and sorting, etc.